### ISSUE TITLE
DMN Drools TCK runner updates

### ISSUE THREAD

**Issue by masskran** [masskran/redhat_bot#486]
Activities on upstream DMN TCK is on hold as no meetings are being held -> no PRs merged on DMN TCK repo.
Hence this PR is pending: https://github.com/dmn-tck/tck/pull/360

In that branch/PR, I am aligning with the original TCK intention that "errorResult" flag is an hint, but cannot be enforced, over the runners.

Example, in case 
* the result from a DMN expression evaluation is null, 
* the TCK expect a null, 
* IFF TCK did not flagged as "errorResult", we no longer just fail simply because the Drools DMN engine reported an error (which was not indicated by the TCK test).
https://github.com/dmn-tck/tck/pull/360/files#diff-49ce6a21913fba22bc06e78fc8fdcaf4R269

Before, if the Drools DMN engine raised an error, but the TCK did not indicated as "errorResult" we were reporting a failure, even if the actual value was consistent with the expected one!

Now is aligned as explained.

Naturally, I've in parallel raised already PR to TCK to fix the test and to provide the proper "errorResult" hint. (https://github.com/dmn-tck/tck/pull/359) 

Once the DMN-TCK-upstream PR for the Drools TCK runner is merged, the commit https://github.com/kiegroup/drools/commit/ca70199cf5edccf5261f153dde3296c646350c61 can be reverted.
